### ISSUE TITLE
Proposal for storyboards template to shorten code needed in prepareForSegue

### ISSUE THREAD

**Issue by iotize** [SwiftGen/SwiftGen#437]
I'm generating code for storyboards using the Swift 4 storyboards template.

There are two steps required to convert from a segue to a `SegueType`:

```swift
override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
  guard let identifier = segue.identifier else {
    return
  }

  guard let segueType = StoryboardSegue.Main(rawValue: identifier) else {
    return
  }

  // perform action based on segue type
}
```

This seems a bit long-winded when repeated across several view controllers.

This is a proposal to include a convenience initialiser in generated `SegueType` enums:

```swift
internal enum StoryboardSegue {
  internal enum Main: String, SegueType {
    case showDetails
    case presentInfo

    init?(from segue: UIStoryboardSegue) {
      guard let identifier = segue.identifier else {
        return nil
      }
      self.init(rawValue: identifier)
    }
  }
}
```

This reduces the above boilerplate to:

```swift
override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
  guard let segueType = StoryboardSegue.Main(from: segue) else {
    return
  }

  // perform action based on segue type
}
```

It's possible to extend each of the enums in my own project to include this, but it seems like a general solution that would be useful in other projects.

I'm happy to include this in a pull request if the change is acceptable, or if there's a better way to do the above, that works too. 🙂

**Comment by djbe** [comment]
Let me point you to this PR: https://github.com/SwiftGen/SwiftGen/pull/364

When that PR gets merged, the could would be:
```swift
override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
    switch TypedStoryboardSegue(segue: segue) {
    case .nextStep(let destination):
        // destination is casted version
    case .login(let destination):
        // destination is casted version
    case .unnamedSegue:
        // This case is added if you have unnamed segues (with identifier "")
    }
}
```
On top of that, it'd be even safer because it would only contain the (possible) segues starting from that controller.

**Comment by iotize** [comment]
Thanks, that looks interesting. Definitely an improvement, better than the suggestion above. What's the likelihood of it being merged soon?

**Comment by djbe** [comment]
Well, I'm pretty sure #419 (split scene/segue) will need to be merged first, and then #364 can be rebased on that. I'd love to release this in the next SwiftGen, but that'll depend on how well developed it'll be.

Any and all feedback on #364 would be very welcome!

**Comment by djbe** [comment]
With the release of SwiftGen 6.0.0, we provide the shorthand initialiser mentioned by @iotize. 

See here how to use it: https://github.com/SwiftGen/SwiftGen/blob/master/Documentation/templates/ib/segues-swift4.md#usage-example
And here's an example of the generated code: https://github.com/SwiftGen/SwiftGen/blob/master/Tests/Fixtures/Generated/IB-iOS/segues-swift4-context-all.swift
